cpprelude/include/cpprelude/iterator.h

```diff
@@ -572,12 +640,15 @@ namespace cpprelude
 	struct const_bucket_array_iterator;
 
 	template<typename T, usize bucket_size>
-	struct bucket_array_iterator
+	struct bucket_array_iterator:std::iterator<std::random_access_iterator_tag, T, isize, T*, T&>
 	{
+		using base = std::iterator<std::random_access_iterator_tag, T, isize, T*, T&>;
 		using data_type = T;
 		T** _bucket_it;
 		T* _element_it;
 		usize _index;
+		usize _bucket_index = 0;
```

Comment:
why these two new member variables?

Reply:
_index was not added by me, I found it already in the code before I start editing it.
As for _bucket_index, that was a debugging variable and I forgot to remove it. sorry, bad me.